imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,186 @@
+            ) from None
+
+    def close(self) -> None:
+        if self._image_file:
+            self._image_file.close()
+
+        self._request.finish()
+
+    def read(self, *, index: int = 0, **kwargs) -> np.ndarray:
+        """Read Raw Image.
+
+        Returns
+        -------
+        nd_image: ndarray
+            The image data
+        """
+
+        nd_image: np.ndarray
+
+        try:
+            nd_image = self._image_file.postprocess(**kwargs)
+        except Exception:
+            pass
+        
+        nd_image = nd_image[None, ...]
+        return nd_image
+
+    def write(
+        self, 
+        ndimage: Union[ArrayLike, List[ArrayLike]]
+    ) -> Optional[bytes]:
+        """RawPy does not support writing.
+        """
+        raise NotImplementedError()
+
+    def iter(self) -> Iterator[np.ndarray]:
+        """Load the image.
+
+        Returns
+        -------
+        nd_image: ndarray
+            The image data
+        """
+
+        try:
+            yield self.read()
+        except Exception:
+            pass
+
+    def metadata(
+        self, index: int = None, exclude_applied: bool = True
+    ) -> Dict[str, Any]:
+        """Read ndimage metadata.
+
+        Parameters
+        ----------
+        exclude_applied : bool
+            If True, exclude metadata fields that are applied to the image while
+            reading. For example, if the binary data contains a rotation flag,
+            the image is rotated by default and the rotation flag is excluded
+            from the metadata to avoid confusion.
+
+        Returns
+        -------
+        metadata : dict
+            A dictionary of format-specific metadata.
+
+        """
+
+        metadata = {}
+
+        ImageSize = self._image_file.sizes
+
+        # TO DO: Find out what information needs to be included below and refine accordingly.
+        metadata["black_level_per_channel"] = self._image_file.black_level_per_channel
+        metadata["camera_white_level_per_channel"] = self._image_file.camera_white_level_per_channel
+        metadata["color_desc"] = self._image_file.color_desc
+        #metadata["color_matrix"] = self._image_file.color_matrix
+        metadata["daylight_whitebalance"] = self._image_file.daylight_whitebalance
+        metadata["dtype"] = self._image_file.raw_image.dtype
+        metadata["flip"] = ImageSize.flip
+        metadata["num_colors"] = self._image_file.num_colors
+        #metadata["raw_colors"] = self._image_file.raw_colors
+        #metadata["raw_colors_visible"] = self._image_file.raw_colors_visible
+        #metadata["raw_image"] = self._image_file.raw_image
+        #metadata["raw_image_visible"] = self._image_file.raw_image_visible
+        #metadata["raw_pattern"] = self._image_file.raw_pattern
+        metadata["raw_type"] = self._image_file.raw_type.name
+        #metadata["rgb_xyz_matrix"] = self._image_file.rgb_xyz_matrix
+        #metadata["tone_curve"] = self._image_file.tone_curve
```

Comment:
Done. But in previous comment, you mentioned that raw_pattern, raw_type and rgb_xyz_matrix are config paramteres. Should we still include them ?

Reply:
Also, after updating the metadata based on all the comments above, it seems only width, height and pixel_aspect are the ones which get included when exclude_applied is set to True. So, in the test function for metadata, I have only included these three since exclude_applied is set to True by default. Please let me know if that is ok. Thanks!

Reply:
Oh we are being trolled by the Github UI 🤣

I left the comment on the last line `"tone_curve"`, but apparently GH always shows a minimum of 4 lines for a comment. I see how that becomes confusing. Please keep `tone_curve` and remove the others :)

Reply:
It's perfectly fine to only assert values of the metadata that is returned by default.

Reply:
Thanks! I have removed others except tone_curve.